docs/content/config/environment.md

```diff
@@ -54,11 +54,11 @@ The Group ID assigned to the static vmail group for `/var/mail` (_Mail storage m
 
 Configures the provisioning source of user accounts (including aliases) for user queries and authentication by services managed by DMS (_Postfix and Dovecot_).
 
-User provisioning via OIDC is planned for the future, see [this tracking issue](https://github.com/docker-mailserver/docker-mailserver/issues/2713).
+User provisioning via OAUTH2 is planned for the future, see [this tracking issue](https://github.com/docker-mailserver/docker-mailserver/issues/2713). OAuth2.0 for machine-to-machine authentication (e.g. Roundcube login) has been implemented (see OAUTH2 section below).
 
 - **empty** => use FILE
 - LDAP => use LDAP authentication
-- OIDC => use OIDC authentication (**not yet implemented**)
+- OAUTH2 => use OAUTH2 authentication (**not yet implemented**)
```

Comment:
Wait - this does not seem to be valid anymore?

ref: https://github.com/docker-mailserver/docker-mailserver/pull/3480/files#diff-845c87c5018d576c8f4654f463f2d2a98a4a536fc2c0491cdc41d2680cc50413L65-L68

Reply:
This PR is only a supplement to a full Auth provider like LDAP/FILE. I'd like to contribute that next if I can.

Reply:
Oh, I read that wrong - all good!

Reply:
I forgot that I removed that. Might it be better to leave it in `start-mailserver.sh` to solidify the difference between the supplement and a full OAUTH2 provider?

Reply:
There is no `OIDC` provider, just a stub that's been lingering for a while 🤷‍♂️ 

Since it's already there, I don't mind leaving it as is, but OAUTH2 is primarily meant to be for authorization AFAIK, with OIDC absorbing/extending that spec into proper authentication support?

Revert:

```suggestion
OIDC => use OIDC authentication (**not yet implemented**)
```

Reply:
I think the core is just auto-configuration using the `.well-known` path. At it's core, it's still OAuth2.0.
I don't have much of a leg to stand on since my knowledge is still somehow minimal despite doing all this work haha. So if you still think OIDC is better, we can change it.